sympy/concrete/summations.py

```diff
@@ -835,11 +836,27 @@ def eval_sum(f, limits):
     value = eval_sum_symbolic(f.expand(), (i, a, b))
     if value is not None:
         return value
+    if not definite:
+        zeta_function = eval_zeta_function(f, (i, a, b))
+        if zeta_function is not None:
+            return zeta_function
     # Do it directly
     if definite:
         return eval_sum_direct(f, (i, a, b))
 
 
+def eval_zeta_function(f, limits):
+    i, a, b = limits
+    n, s = Wild('n'), Wild('s')
+    result = f.match(n ** (-s))
+    if result is not None and b == S.Infinity and result[n].has(i):
+        if result[n].is_Add:
```

Comment:
It could also be something like `i**2 + 1`. I think you should rather match against `(a*n + b)**s` where `a`, `b` and `s` exclude `i`.